src/pseudopeople/noise_functions.py

```diff
@@ -333,17 +334,20 @@ def choose_wrong_options(
     options = selection_options.loc[selection_options[selection_type].notna(), selection_type]
     new_values = vectorized_choice(
         options=options,
-        n_to_choose=len(data),
+        n_to_choose=len(to_noise_idx),
         randomness_stream=randomness_stream,
         additional_key=f"{column_name}_incorrect_select_choice",
     ).to_numpy()
 
-    return pd.Series(new_values, index=data.index, name=column_name)
+    data.loc[to_noise_idx, column_name] = ensure_dtype(
```

Comment:
Why do we need to call `ensure_dtype` again here? Is it possible for this to change the dtype if the options are not "wrong"?

Reply:
I think in this case it is not possible, but I think it is simpler to reason about if we always explicitly ensure the dtype. The behavior of pandas when you set part of a Series with values that do not have the same dtype is pretty complex, so I think it's easier to explicitly state that we are _not_ doing that.